src/PhpParser/Visitor/ParentConnectorVisitor.php

```diff
@@ -59,20 +57,23 @@ public function beforeTraverse(array $nodes): ?array
         return null;
     }
 
-    public function enterNode(Node $node): ?Node
+    public function afterTraverse(array $nodes): ?array
+    {
+        return null;
+    }
+
+    public function enterNode(Node $node)
     {
         $stackCount = count($this->stack);
 
-        if ($stackCount !== 0) {
-            $node->setAttribute(self::PARENT_KEY, $this->stack[$stackCount - 1]);
-        }
+        ParentConnector::setParent($node, $this->stack[$stackCount - 1] ?? null);
```

Comment:
I like the explicit `if ($stackCount !== 0) {}` better. Easier to read.

Reply:
The difference here is not the count check, but that the attribute is set (to null) even if no parent is found

Reply:
So there are cases where a parent has to be unregistered, is this right?

Reply:
In practice it doesn't make any difference – when it was not set we default to `null` from `getAttribute()`, but i thought that we might as well have all our "metadata" available so that upon inspection you see clearly everything